src/test/lib/async-replace_test.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/// <reference path="../../../node_modules/@types/mocha/index.d.ts" />
+/// <reference path="../../../node_modules/@types/chai/index.d.ts" />
+
+import {asyncReplace} from '../../lib/async-replace.js';
+import {html, render} from '../../lit-html.js';
+
+import {TestAsyncIterable} from './test-async-iterable.js';
+
+const assert = chai.assert;
+
+// Set Symbol.asyncIterator on browsers without it
+if (typeof Symbol !== undefined && Symbol.asyncIterator === undefined) {
+  Object.defineProperty(Symbol, 'asyncIterator', {value: Symbol()});
+}
+
+suite('asyncReplace', () => {
+
+  let container: HTMLDivElement;
+  let iterable: TestAsyncIterable<string>;
+
+  setup(() => {
+    container = document.createElement('div');
+    iterable = new TestAsyncIterable<string>();
+  });
+
+  test('renders an async iterable', async () => {
```

Comment:
This actually tests a very particular rendering strategy. It's more like, "replaces content as the async iterable yields new values".

Reply:
fixed